.github/workflows/doxygen-gh-pages.yml

```diff
@@ -0,0 +1,39 @@
+name: Deploy Doxygen results to Github Pages
+
+on: push
```

Comment:
shouldn't this only run when pushing to `master`?

Reply:
yes. it's for testing (on each push)

Reply:
well, it should only work when a new commit is pushed to `master`